From d8a9cd0719635ac10749b13e248e6543e62aa2ef Mon Sep 17 00:00:00 2001
From: Xi Ruoyao <xry111@xry111.site>
Date: Wed, 10 Dec 2025 15:34:32 +0800
Subject: [PATCH 59/63] loonggpu: Drop the call to drm_fb_helper_alloc_info

On Linux >= 6.19 it's called by the DRM generic code and no longer
exported for the drivers.

Link: https://git.kernel.org/torvalds/c/63c971af4036
Signed-off-by: Xi Ruoyao <xry111@xry111.site>
---
 include/loonggpu_helper.h | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/include/loonggpu_helper.h b/include/loonggpu_helper.h
index 82cd586..90132af 100644
--- a/include/loonggpu_helper.h
+++ b/include/loonggpu_helper.h
@@ -527,7 +527,9 @@ static inline void lg_ttm_bo_unlock_delayed_workqueue(struct loonggpu_device *ad
 
 static inline struct fb_info *lg_drm_fb_helper_alloc_info(struct drm_fb_helper *helper)
 {
-#if defined(LG_DRM_FB_HELPER_ALLOC_INFO)
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(6, 19, 0)
+	return helper->info;
+#elif defined(LG_DRM_FB_HELPER_ALLOC_INFO)
 	return drm_fb_helper_alloc_info(helper);
 #else
 	return drm_fb_helper_alloc_fbi(helper);
-- 
2.52.0

